Improve transforms and use new LINQ-like syntax

Tom94 opened the pull request:
- [x] Depends on https://github.com/ppy/osu-framework/pull/902

Adjusts the project to make use of the new LINQ-like transform chaining syntax. Right now, only very few transform usages have been migrated, and the plan is to go further step-by-step.

This PR also slightly improves the way rolling counter generic transforms work.

Tom94 committed:
Use TransformTo in rolling counters

Tom94 committed:
Merge branch 'master' of https://github.com/ppy/osu into better-transforms

Tom94 committed:
Nicer generic rolling counters

Tom94 committed:
Update framework and fix compilation

Most issues were related to BeginLoopedSequence usage and lack of
"this." in front of transform helpers.

Tom94 committed:
Remove some unnecessary usings

Tom94 committed:
Update framework

Tom94 committed:
Delay -> ApplyDelay; DelayReset -> ResetDelay

Tom94 committed:
Replace several usages of BeginDelayedSequence with LINQ-style

Tom94 committed:
ApplyDelay -> AddDelay

Tom94 committed:
Update framework

Tom94 committed:
Try different formatting style

smoogipooo commented on osu.Game/Screens/Menu/OsuLogo.cs:
```diff
@@ -239,21 +239,30 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
-            logoBeatContainer.ScaleTo(1 - 0.02f * amplitudeAdjust, early_activation, EasingTypes.Out)
-                .Then().ScaleTo(1, beatLength * 2, EasingTypes.OutQuint);
+            logoBeatContainer
+                .ScaleTo(1 - 0.02f * amplitudeAdjust, early_activation, EasingTypes.Out)
```
I really dislike this being on a new line by itself. Can we use
```
logoBeatContainer.ScaleTo()
                 .Then()
                 .ScaleTo()
```
?

Tom94 committed:
Get rid of raw AddDelay calls within osu.Game

Tom94 committed:
Get rid of AddDelay & ResetDelay in osu.Game.Rulesets.Osu

Tom94 committed:
Remove remaining usages of AddDelay

Tom94 committed:
Merge branch 'master' of https://github.com/ppy/osu into better-transforms

# Conflicts:
#	osu-framework
#	osu.Game/Graphics/Cursor/OsuTooltipContainer.cs
#	osu.Game/Overlays/DragBar.cs

Tom94 committed:
Update framework

Tom94 commented:
The most recent framework improvements now also fix unintuitive overriding of transforms such as was seen in the tooltip container (this fix e.g. is no longer needed: https://github.com/ppy/osu/pull/1024 ).

As a nice side effect, some other bugs have disappeared, such as the main menu staying expanded when rapidly pressing ESC after clicking the osu logo. The following video demonstrates the fix.
https://streamable.com/a9mqf

Tom94 committed:
Fix osu logo shockwave playing when transition is aborted

This is a very nice use-case scenario for TransformSequence.OnComplete

Tom94 commented:
The last commit is a nice demonstration of the `OnComplete` feature of `TransformSequence`. The passed action is only invoked if the sequence completes successfully, i.e. none of its transforms gets cancelled or overwritten by another transform. In this particular case it allows the impact of the osu logo's intro sequence to only play if the transition is not interrupted; see https://streamable.com/oo0vo .

Tom94 committed:
Ensure toolbar disappears when osu logo transition is cancelled

Tom94 committed:
Merge branch 'master' of github.com:ppy/osu into better-transforms

# Conflicts:
#	osu-framework

Tom94 committed:
Merge branch 'master' of github.com:ppy/osu into better-transforms

# Conflicts:
#	osu-framework

Tom94 committed:
Update framework

Tom94 committed:
Update framework

Tom94 committed:
Update dependency caching according to framework

Tom94 committed:
Minor refactor of TestCaseBeatSyncedContainer

Tom94 committed:
Merge branch 'master' into better-transforms

Tom94 committed:
Merge branch 'master' of github.com:ppy/osu into prevent-dependency-pollution

Tom94 committed:
Update framework

Tom94 committed:
Don't obtain DependencyContainer via DI

Tom94 committed:
Don't obtain DependencyContainer via DI

Tom94 committed:
Address CI concerns and update framework

Tom94 committed:
Merge branch 'prevent-dependency-pollution' of github.com:Tom94/osu into better-transforms

Tom94 committed:
Fix broken test case

Tom94 committed:
Merge branch 'better-transforms' of github.com:Tom94/osu into prevent-dependency-pollution

Tom94 committed:
Update framework

Tom94 commented:
Requires a disabled ReSharper warning on a purely transformed member.

peppy committed:
Supress compiler warning

I don't think this is a sustainable solution if we start doing this in more places.

peppy commented:
Fomatting looks good. Also resharper's reformat code handles it all correctly, so no qualms here 👍
